lib/performance/metrics/metric_sender.rb

```diff
@@ -10,10 +10,10 @@ class MetricSender
     VALID_PERIODS = %w[week day month].freeze
     VALID_STATS = %w[active_users completion_rate roaming_users volumetrics].freeze
     STATS = {
-      active_users: PerformancePlatform::Gateway::ActiveUsers,
-      completion_rate: PerformancePlatform::Gateway::CompletionRate,
-      roaming_users: PerformancePlatform::Gateway::RoamingUsers,
-      volumetrics: PerformancePlatform::Gateway::Volumetrics,
+      active_users: Performance::UseCase::ActiveUsers,
+      completion_rate: Performance::UseCase::CompletionRate,
+      roaming_users: Performance::UseCase::RoamingUsers,
+      volumetrics: Performance::UseCase::Volumetrics,
```

Comment:
Why are these changing from Gateways to UseCases? As they are getting the data isn't Gateway the right type for them?

Separately the comment above could do with updating to avoid future confusion. (it refers to `PerformancePlatform::Gateway::ActiveUsers`)

Reply:
I thought because they use a repository to get their data they would not be a gateway but more a use case. Gateway sounds more low level access to me. what do you think?

Reply:
I think in my head the repository is the Gateway _Implementation_... Though I guess a UseCase of `[Get]ActiveUsers` is valid too. Did I mention this confuses me? 😆 

![](https://github.com/madetech/clean-architecture/raw/master/ruby/clean-architecture.png)